File: frontend/demo/component/upload/upload-error-messages.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/form-layout';
import '@vaadin/upload';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import type { FormLayoutResponsiveStep } from '@vaadin/form-layout';
import { applyTheme } from 'Frontend/generated/theme';
/* prettier-ignore */
import { createFakeFilesUploadErrorMessagesA, createFakeFilesUploadErrorMessagesB } from './upload-demo-mock-files'; // hidden-source-line

const layoutSteps: FormLayoutResponsiveStep[] = [
  { minWidth: 0, columns: 1, labelsPosition: 'top' },
  { minWidth: '540px', columns: 2, labelsPosition: 'top' },
];

@customElement('upload-error-messages')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  private cautionI18n = {
    uploading: {
      error: {
        unexpectedServerError: 'Unexpected Server Error',
      },
    },
  };

  // tag::snippet[]
  private recommendedI18n = {
    uploading: {
      error: {
        unexpectedServerError: "File couldn't be uploaded, try again later",
      },
    },
  };

  protected override render() {
    return html`
      <!-- end::snippet[] -->
      <vaadin-form-layout .responsiveSteps="${layoutSteps}">
        <div>
          <strong>Caution</strong>
          <vaadin-upload
            id="upload-caution"
            nodrop
            .i18n="${this.cautionI18n}"
            .files="${createFakeFilesUploadErrorMessagesA() /* hidden-source-line */}"
          ></vaadin-upload>
        </div>
        <div>
          <strong>Recommended</strong>
          <!-- tag::snippet[] -->
          <vaadin-upload
            id="upload-recommended"
            nodrop
            .i18n="${this.recommendedI18n}"
            .files="${createFakeFilesUploadErrorMessagesB() /* hidden-source-line */}"
          ></vaadin-upload>
          <!-- end::snippet[] -->
        </div>
      </vaadin-form-layout>
      <!-- tag::snippet[] -->
    `;
  }
  // end::snippet[]
}

File: src/main/java/com/vaadin/demo/component/upload/UploadErrorMessages.java
package com.vaadin.demo.component.upload;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.flow.component.formlayout.FormLayout;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.html.NativeLabel;
import com.vaadin.flow.component.upload.Upload;
import com.vaadin.flow.component.upload.receivers.MultiFileMemoryBuffer;
import com.vaadin.flow.router.Route;

@Route("upload-error-messages")
public class UploadErrorMessages extends Div {

    public UploadErrorMessages() {
        Div cautionExample = setupCautionExample();
        Div recommendedExample = setupRecommendedExample();

        FormLayout formLayout = new FormLayout(cautionExample,
                recommendedExample);
        formLayout.setResponsiveSteps(new FormLayout.ResponsiveStep("0", 1),
                new FormLayout.ResponsiveStep("540px", 2));

        add(formLayout);
    }

    private Div setupCautionExample() {
        MultiFileMemoryBuffer buffer = new MultiFileMemoryBuffer();
        Upload upload = new Upload(buffer);
        upload.setDropAllowed(false);

        UploadExamplesI18N i18N = new UploadExamplesI18N();
        i18N.getUploading().getError()
                .setUnexpectedServerError("Unexpected Server Error");
        upload.setI18n(i18N);

        NativeLabel label = new NativeLabel("Caution");
        label.getStyle().set("font-weight", "600");
        upload.setId("upload-caution");
        label.setFor(upload.getId().get());
        upload.getElement().callJsFunction("setupMockErrorResponse"); // hidden-source-line
        upload.getElement().executeJs( // hidden-source-line
                "this.files = this.createFakeFilesUploadErrorMessagesA()"); // hidden-source-line

        return new Div(label, upload);
    }

    private Div setupRecommendedExample() {
        MultiFileMemoryBuffer buffer = new MultiFileMemoryBuffer();
        // tag::snippet[]
        Upload upload = new Upload(buffer);
        upload.setDropAllowed(false);

        UploadExamplesI18N i18N = new UploadExamplesI18N();
        i18N.getUploading().getError().setUnexpectedServerError(
                "File couldn't be uploaded, please try again later");
        upload.setI18n(i18N);
        // end::snippet[]

        NativeLabel label = new NativeLabel("Recommended");
        label.getStyle().set("font-weight", "600");
        upload.setId("upload-recommended");
        label.setFor(upload.getId().get());
        upload.getElement().callJsFunction("setupMockErrorResponse"); // hidden-source-line
        upload.getElement().executeJs( // hidden-source-line
                "this.files = this.createFakeFilesUploadErrorMessagesB()"); // hidden-source-line

        return new Div(label, upload);
    }

    public static class Exporter // hidden-source-line
            extends DemoExporter<UploadErrorMessages> { // hidden-source-line
    } // hidden-source-line
}
